src/lib.rs

```diff
@@ -61,17 +74,20 @@ where
     fn write_byte(&mut self, mut data: u8) {
         for _ in 0..8 {
             if (data & 0x80) != 0 {
-                block!(self.timer.wait()).ok();
-                self.pin.set_high();
-                block!(self.timer.wait()).ok();
-                block!(self.timer.wait()).ok();
-                self.pin.set_low();
+                //1bit, 666ns on time, 666ns off time
```

Comment:
The low block was outside done this way, so that the iterator can run during the low time without needing additional overhead

Reply:
Additionally, doing it your way may cut the time for the high block short (since the timer runs while low)